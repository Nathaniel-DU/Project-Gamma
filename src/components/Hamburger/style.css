/* Position and sizing of burger button */
.bm-burger-button {
    position: fixed;
    width: 36px;
    height: 30px;
    left: 36px;
    top: 36px;
  }
  
  /* Color/shape of burger icon bars */
  .bm-burger-bars {
    background: #ADEFD1FF;
    height: 100%;
  }
  
  /* Position and sizing of clickable cross button */
  .bm-cross-button {
    height: 24px;
    width: 24px;
  }
  
  /* Color/shape of close button cross */
  .bm-cross {
    background: #bdc3c7;
  }
  
  /*
  Sidebar wrapper styles
  Note: Beware of modifying this element as it can break the animations - you should not need to touch it in most cases
  */
  .bm-menu-wrap {
    position: fixed;
    height: 100%;
  }
  
  /* General sidebar styles */
  .bm-menu {
    background: #ADEFD1FF;
    padding: 1em 1.5em 0;
    font-size: 1.15em;
    overflow: hidden;
  }
  
  
  /* Wrapper for item list */
  .bm-item-list {
    color: #000000;
  }
  
  /* Individual item */
  .bm-item {
    display: inline-block;
  }
  
  /* Styling of overlay */
  .bm-overlay {
    background: rgba(0, 0, 0, 0.3);
  }

  :focus {
    outline: none;
}
