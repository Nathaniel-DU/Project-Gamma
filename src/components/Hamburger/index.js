import React from 'react';
import { stack as Menu } from 'react-burger-menu'
import Signout from '../Signout';
import './style.css';

class Hamburger extends React.Component {
  showSettings (event) {
    event.preventDefault();
  }

  render () {
    return (
        <div>
            <Menu pageWrapId={"page-wrap"} outerContainerId={ "outer-container" }>
                {
                  // <Signout/>
                }
            </Menu>
        </div>
    );
  }
}

export default Hamburger;