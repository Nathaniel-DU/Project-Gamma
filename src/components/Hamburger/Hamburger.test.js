import React from 'react';
import { shallow } from 'enzyme';
import Hamburger from './index';

describe(`Hamburger`, () => {
    it(`should render correctly in "debug" mode`, () => {
        const component = shallow(<Hamburger debug />);
        expect(component).toMatchSnapshot();
    });
});