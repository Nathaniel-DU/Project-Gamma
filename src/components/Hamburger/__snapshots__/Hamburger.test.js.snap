// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger should render correctly in "debug" mode 1`] = `
<div>
  <Menu
    bodyClassName=""
    burgerBarClassName=""
    burgerButtonClassName=""
    className=""
    crossButtonClassName=""
    crossClassName=""
    disableAutoFocus={false}
    disableCloseOnEsc={false}
    htmlClassName=""
    id=""
    itemClassName=""
    itemListClassName=""
    menuClassName=""
    morphShapeClassName=""
    noOverlay={false}
    noTransition={false}
    onIconHoverChange={[Function]}
    onStateChange={[Function]}
    outerContainerId="outer-container"
    overlayClassName=""
    pageWrapId="page-wrap"
    styles={Object {}}
    width={300}
  >
    <div>
      <Logout />
      <EditProfileButton />
      <FriendSearch />
      <h4>
        Friends:
      </h4>
      <hr />
      <Friends />
    </div>
  </Menu>
</div>
`;
